src/androidTest/java/com/couchbase/lite/support/BatcherTest.java

```diff
@@ -285,6 +282,14 @@ public void process(List<String> itemsToProcess) {
             success = latch1.await(500, TimeUnit.MILLISECONDS);
             assertTrue(success);
 
+            // NOTE: https://github.com/couchbase/couchbase-lite-java-core/issues/518
+            //       CountDownLatch became 0, but task might not be done yet.
+            //       Give little bit of time for slower environment.
+            try {
+                Thread.sleep(500);
+            } catch (Exception ex) {
+            }
+
             // add another object
             batcher.queueObject(new String());
```

Comment:
I removed my previous comments to avoid confusion.

1. I think you can simulate the issue by adding some delays at the end of processNow method (L222).

2. As there are some items in the inbox, should it reschedule the processing instead of returning immediately? 
https://github.com/couchbase/couchbase-lite-java-core/blob/b37a40b4786292246aa81c3517b531de0dc83c3b/src/main/java/com/couchbase/lite/support/Batcher.java#L235-L237